Fix: Handle dict objects in Anthropic streaming response

jgowdy-godaddy opened the pull request:
Fixes #10792 

## Summary
- Fixed an issue where dictionary objects in Anthropic streaming responses were not properly converted to SSE format strings
- When using the Anthropic  endpoint in streaming mode, the function was directly yielding dictionary objects
- Added proper type checking and formatting for dictionary objects in the streaming response

## Test plan
- Start the LiteLLM proxy server
- Send a POST request to  with streaming enabled
- Verify that the streaming response works correctly and doesn't throw AttributeError

jgowdy-godaddy committed:
fix: handle dict objects in Anthropic streaming response

Fix issue where dictionary objects in Anthropic streaming responses
were not properly converted to SSE format strings before being yielded,
causing AttributeError: 'dict' object has no attribute 'encode'

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/BerriAI/litellm?pullRequest=11032) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/BerriAI/litellm?pullRequest=11032) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/BerriAI/litellm?pullRequest=11032) it.</sub>

ishaan-jaff commented on litellm/proxy/anthropic_endpoints/endpoints.py:
```diff
@@ -40,7 +40,12 @@ async def async_data_generator_anthropic(
+            # Convert dictionary to a properly formatted SSE string for streaming
+            if isinstance(chunk, dict):
+                chunk_str = json.dumps(chunk)
+                yield f"data: {chunk_str}\n\n"
```
place `data` in constants.py near `STREAM_SSE_DONE_STRING`

ishaan-jaff commented on litellm/proxy/anthropic_endpoints/endpoints.py:
```diff
@@ -40,7 +40,12 @@ async def async_data_generator_anthropic(
 
-            yield chunk
+            # Convert dictionary to a properly formatted SSE string for streaming
+            if isinstance(chunk, dict):
```
can this be a small helper called return_anthropic_chunk

ishaan-jaff commented on litellm/proxy/anthropic_endpoints/endpoints.py:
```diff
@@ -40,7 +40,12 @@ async def async_data_generator_anthropic(
-            yield chunk
+            # Convert dictionary to a properly formatted SSE string for streaming
+            if isinstance(chunk, dict):
+                chunk_str = json.dumps(chunk)
```
use safe_dumps in litellm/litellm_core_utils/safe_json_dumps.py

ishaan-jaff reviewed (changes requested):
Thanks, reviewed. Please add a test tests/litellm/proxy/anthropic_endpoints/test_endpoints.py

jgowdy-godaddy committed:
fix: refactor Anthropic streaming response handling

- Added STREAM_SSE_DATA_PREFIX constant in constants.py
- Created return_anthropic_chunk helper function for better maintainability
- Using safe_dumps from safe_json_dumps.py for improved JSON serialization
- Added unit test for dictionary object handling in streaming response

jgowdy-godaddy committed:
fix: correct patch path in anthropic_endpoints test

ishaan-jaff commented:
@jgowdy-godaddy were you able to see the original issue ? I'm on an older version and don't see the issue 

Tried this curl command 

```
curl -X POST http://localhost:4000/v1/messages \
  -H "content-type: application/json" \
  -H "accept: application/json" \
  -H "Authorization: Bearer sk-1234" \
  -d '{
    "model": "claude-3-5-haiku-20241022",
    "max_tokens": 512,
    "messages": [
      {
        "role": "user",
        "content": "Hi"
      }
    ],
    "system": [
      {
        "type": "text",
        "text": "You are an AI Agent"
      }
    ],
    "temperature": 0,
    "stream": true
  }'
``
